### ISSUE TITLE
[BUG] symbols in non-Main modules get cached as Main's

### ISSUE THREAD

**Issue by aviatesk** [JunoLab/Juno.jl#407]
[`updateeditor`](https://github.com/JunoLab/Atom.jl/blob/master/src/goto.jl#L244-L251) can be called with `mod == "Main"` even when `path` is really in non-`Main` module.

This happens when we open/switch to an editor (in non-`Main` module, say module `Mod`) -- `updateeditor` handler can be called before `module` handler detects the editor is in `Mod`.

And this causes duplicated (and even worse, wrong) caches, and end up eating some memory.

***

Maybe adding an debounce [the call of `updateEditor`](https://github.com/JunoLab/atom-julia-client/blob/avi/renamerefactor/lib/runtime/outline.js#L63-L65) would address this issue.

**Comment by aviatesk** [comment]
yeah, that should be the fundamental fix for this, and I agree with that we should fix that with high priority.
I will file an issue for that.

---

Still, I want to address this soon even with a temporary fix... It's very annoying for me, who works on fairly low spec PC.
And now I found there is actually no need to update symbols cache on editor switch -- we only want to update symbols [when the source is modified](https://github.com/JunoLab/atom-julia-client/blob/avi/renamerefactor/lib/runtime/outline.js#L46-L50).

So, I will make an PR, so that we don't run `updatesymbols` on `onDidChangeActiveTextEditor`.

**Comment by aviatesk** [comment]
And the code that updates symbols/outline on editor changes (i.e. `onDidStopChanging`) is [already debounced](https://flight-manual.atom.io/api/v1.30.0/TextEditor/#instance-onDidStopChanging) and only occurs after the editor module is resolved, so this issue would be addressed by that, for now.

**Comment by pfitzseb** [comment]
Actually, isn't `mod == nothing` in the `updateeditor` handler if we don't know the module because the handlers got called in the "wrong" order? Could just bail out early in that case instead of falling back to `Main`.

**Comment by aviatesk** [comment]
I agree with that if `mod == nothing`, then the symbols cache would be really safe.

***

[This](https://github.com/JunoLab/atom-julia-client/blob/avi/renamerefactor/lib/runtime/outline.js#L81)  and [this](https://github.com/JunoLab/Atom.jl/blob/avi/renamerefactor/src/outline.jl#L78) would fall back to `Main`, but even if we remove the both fallback code, still `modules.current()` keeps returning the `"Main"` for that case.

So we need to modify modules.coffee as well. I think we would end up addressing this with https://github.com/JunoLab/Juno.jl/issues/411 anyway ?

**Comment by aviatesk** [comment]
So this issue would stay half-fixed until we really fix our module detection, if I understand correctly.
